src/meta/src/controller/utils.rs

```diff
@@ -220,6 +222,22 @@ where
     Ok(())
 }
 
+/// `check_user_name_duplicate` checks whether the user is already existed in the cluster.
+pub async fn check_user_name_duplicate<C>(name: &str, db: &C) -> MetaResult<()>
```

Comment:
BTW, do we also have a unique key on the `user`'s `name`?

Reply:
Yes, we have. Checking duplicate here is only for better error reporting.

Reply:
LGTM!